### PR TITLE
Fix retrieving images from Google Container Registry

### LINKED ISSUE #39: docker2aci fails to retrieve images from Google Container Registry
As noted in coreos/rkt#726:

```
$ ./docker2aci docker://gcr.io/google_containers/kibana:1.2


Error: conversion error: error getting ImageID from tag 1.2: HTTP code: 405. URL: https://gcr.io/v1/repositories/google_containers/kibana/tags/1.2
```

From my very brief look, it seems like gcr.io doesn't properly implement the Docker registry API? Hence the command above failing - the tags endpoints (e.g. https://gcr.io/v1/repositories/google_containers/kibana/tags, https://gcr.io/v1/repositories/google_containers/kibana/tags/1.2) are 405ing for no good reason I can tell...

But Docker works just fine:
```
$ docker pull gcr.io/google_containers/kibana:1.2
Pulling repository gcr.io/google_containers/kibana
46724d0d1108: Pulling dependent layers 
511136ea3c5a: Download complete 
fa4fd76b09ce: Downloading [==================================================>] 66.26 MB/66.26 MB
```

### PR DESCRIPTION
**Pull request by iaguis**
Google Container Registry doesn't implement the Docker registry API
correctly: when trying to access
"REGISTRY/repositories/IMAGE/NAME/tags/TAG" it fails with a 405.

Instead of getting the tag from that URL, get the JSON with all the tags
and look for the one we want. This is what Docker does.

Fixes #39

### CODE REVIEW

**Comment by jonboulle** [comment]
@iaguis I'm confused how this isn't a bug in GCR tbh; are they just relying on the Docker client to fall back to retrieving all tags? :-/

File: lib/backend/repository/repository.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func getRepoData(indexURL string, remote string) (*RepoData, error) {
 
 func getImageIDFromTag(registry string, appName string, tag string, repoData *RepoData) (string, error) {
 	client := &http.Client{}
-	req, err := http.NewRequest("GET", "https://"+path.Join(registry, "repositories", appName, "tags", tag), nil)
```
Comment (jonboulle) [inline lib/backend/repository/repository.go]: Would you mind adding a comment explaining this for posterity?

**Comment by jonboulle** [comment]
lgtm

**Comment by iaguis** [comment]
> @iaguis I'm confused how this isn't a bug in GCR tbh; are they just relying on the Docker client to fall back to retrieving all tags? :-/

I think so. It's clearly [in the API](https://docs.docker.com/reference/api/registry_api/#get-image-id-for-a-particular-tag) :/

File: lib/backend/repository/repository.go (hunk 1 above)
Comment (iaguis) [inline lib/backend/repository/repository.go]: Sure

**Comment by iaguis** [comment]
Thanks!

**Pull request merged**